### PR TITLE
Users api

### PR DESCRIPTION
**Pull request by chaudharyprakash**

### CODE REVIEW

File: app/controllers/api/v1/omniauth_callbacks_controller.rb [hunk 1]
```diff
@@ -0,0 +1,28 @@
+class Api::V1::OmniauthCallbacksController < Devise::OmniauthCallbacksController
```
Comment (charusat09) [inline app/controllers/api/v1/omniauth_callbacks_controller.rb:1]: @chaudharyprakash Why do we need this classes??

File: app/controllers/api/v1/base_controller.rb [hunk 2]
```diff
@@ -0,0 +1,16 @@
+class Api::V1::BaseController < ApplicationController
+  skip_before_action  :verify_authenticity_token
+
+  def authorize_user!
+    if params[:auth_token].present?  
+      @token = Authentication.find_by(auth_token: params[:auth_token])
+      if token.present?
+        @current_user = token.user
```
Comment (charusat09) [inline app/controllers/api/v1/base_controller.rb:8]: @chaudharyprakash will you please use @token here. It's my bad.

File: app/controllers/api/v1/registrations_controller.rb [hunk 3]
```diff
@@ -0,0 +1,58 @@
+class Api::V1::RegistrationsController < Api::V1::BaseController
+  before_action :set_user, only: [:edit, :update,:destroy]
```
Comment (KanjariyaHardik) [inline app/controllers/api/v1/registrations_controller.rb]: :edit  -> This value is of which method ?, Should be removed if method is not there

File: app/models/authentication.rb [hunk 4]
```diff
@@ -0,0 +1,4 @@
+class Authentication < ApplicationRecord
+	belongs_to :user
```
Comment (KanjariyaHardik) [inline app/models/authentication.rb]: Spacing Issue, 2 space

File: app/serializers/user_serializer.rb [hunk 5]
```diff
@@ -0,0 +1,8 @@
+class UserSerializer < ActiveModel::Serializer
+	attributes :id, :name, :email, :auth_token
+	def auth_token
+		object.authentications.each do |user|
+			user.auth_token
+		end
+	end
+end
```
Comment (KanjariyaHardik) [inline app/serializers/user_serializer.rb:8]: Indentation issue in whole class

File: app/assets/javascripts/api/v1/base.coffee [hunk 6]
```diff
@@ -0,0 +1,3 @@
+# Place all the behaviors and hooks related to the matching controller here.
+# All this logic will automatically be available in application.js.
+# You can use CoffeeScript in this file: http://coffeescript.org/
```
Comment (KanjariyaHardik) [inline app/assets/javascripts/api/v1/base.coffee]: Remove this file, If we won't use this file

File: app/assets/stylesheets/api/v1/base.scss [hunk 7]
```diff
@@ -0,0 +1,3 @@
+// Place all the styles related to the api/v1/base controller here.
+// They will automatically be included in application.css.
+// You can use Sass (SCSS) here: http://sass-lang.com/
```
Comment (KanjariyaHardik) [inline app/assets/stylesheets/api/v1/base.scss]: Remove this file, If we won't use this file

**Comment by charusat09** [comment]
Done

**Pull request closed without merging**